api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java

```diff
@@ -78,6 +87,37 @@ public void shouldSaveNonExistingClaimant(EligibilityStatus eligibilityStatus, C
         verify(converter).convert(claimDTO);
     }
 
+    @Test
+    void shouldAddVoucherEntitlementToClaim() {
+        //given
+        Claimant claimant = aValidClaimantBuilder().build();
+        Claim claim = buildClaim(claimant);
+        ClaimDTO claimDTO = aValidClaimDTO();
+        VoucherEntitlement voucherEntitlement = aValidVoucherEntitlement();
+        given(converter.convert(any())).willReturn(claim);
+        given(claimantRepository.eligibleClaimExistsForNino(any())).willReturn(false);
+        EligibilityResponse eligibilityResponse = anEligibilityResponse();
+        given(client.checkEligibility(any())).willReturn(eligibilityResponse);
+        given(eligibilityStatusCalculator.determineEligibilityStatus(any())).willReturn(ELIGIBLE);
+        given(entitlementCalculator.calculateVoucherEntitlement(any(), any())).willReturn(voucherEntitlement);
+
+        //when
+        Claim result = claimService.createClaim(claimDTO);
+
+        //then
+        assertThat(result).isEqualTo(claim);
+        assertThat(result.getVoucherEntitlement()).isEqualTo(voucherEntitlement);
+        Claimant expectedClaimant = buildExpectedClaimant(claimant, ClaimStatus.NEW, ELIGIBLE);
```

Comment:
Would it also be worthwhile checking that the Claimant on the result is equal to this? Otherwise if you're just using this to verify what's saved, then I'd rename it to `expectedSavedClaimant` for clarity.